tests/TestCase/QueryBuilderTest.php

```diff
@@ -732,4 +732,35 @@ public function testParseNot()
         ];
         $this->assertEquals($expected, $filter);
     }
+
+    /**
+     * Tests the parse() method with numerically indexed arrays
+     *
+     * @return void
+     */
+    public function testParseNumericArray()
+    {
+        $builder = new QueryBuilder();
+        $filter = $builder->parse([
+            $builder->simpleQueryString('name', 'mark'),
+            ['age >' => 29],
+            'not' => [
+                ['name' => 'jose'],
+                ['age >' => 35],
+            ],
+        ]);
+        $expected = [
+            $builder->simpleQueryString('name', 'mark'),
+            $builder->and(
+                $builder->gt('age', 29),
```

Comment:
```suggestion
                $builder->gt('age', 29)
```

This is causing a syntax error on PHP 7.2